src/main/java/com/google/devtools/build/lib/analysis/RunfilesSupport.java

```diff
@@ -95,7 +97,7 @@ private static RunfilesSupport create(
       RuleContext ruleContext, Artifact executable, Runfiles runfiles, CommandLine args) {
     Artifact owningExecutable = Preconditions.checkNotNull(executable);
     boolean createManifest = ruleContext.getConfiguration().buildRunfilesManifests();
-    boolean createSymlinks = createManifest && ruleContext.getConfiguration().buildRunfiles();
+    boolean createSymlinks = ruleContext.getConfiguration().shouldCreateRunfilesTree();
```

Comment:
With the exception of this line, the rest of the changes in this file are refactorings that are independent of the logic change.
Could you create a separate PR just for refactorings? That'd make the logic-changing parts easier to review.

Reply:
done (inside google)